Sources/Apodini/Response/Status.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  Status.swift
+//  
+//
+//  Created by Paul Schmiedmayer on 2/3/21.
+//
+
+import NIO
+
+
+/// An `Status` expresses additional information that can be passed to a `Response`
+public enum Status: ResponseTransformable {
+    /// The request was handled and the response contains the expected content
+    case ok
+    /// The request was handled and a new resource has been created
+    case created
+    /// The request was handled and the response does not contain any content
```

Comment:
Maybe we could add a `delete` case here, although I think this is normally already handled by 200 or 204. Im not 100% sure, if this falls more into the error section, but what about an 304, if an update did not succeed

Reply:
If we think about HTTP status codes this is already handled by the ones named above. Maybe we should generally think about more general status codes (e.g. `success` instead of `ok`) and some additional expressions that might be eventually matched to the same status code. For now I would suggest to remain with the three here as the `deleted` case would easily be covered with the `ok` `Status`.

An update that has been failed should be indicated by an error code. HTTP Status code 304 is mostly used to indicate that the resource requested did not change since the last request and the cache on the client is still valid (as e.g. indicated by the ETag header value). I think that a failed update is already handled by the Apodini Error. Nevertheless this might be something that the `Database` handlers might want to indicate using the `@Throws` property wrapper.